tests/integration/Owncloud/OcisPhpSdk/OcisResourceTest.php

```diff
@@ -530,4 +530,61 @@ public function testResourcePreview(): void
             }
         }
     }
+
+    public function testFileGetResourcesMetadata(): void
+    {
+        $resourceMetadata = $this->personalDrive->getResourceMetadata('/somefile.txt');
+        $driveResources = $this->personalDrive->getResources();
+        foreach ($driveResources as $driveResource) {
+            if ($driveResource->getName() === $resourceMetadata['name']) {
+                $this->assertEquals($driveResource->getId(), $resourceMetadata['id'], "Expected file id to match, but they are different");
+                $this->assertEquals($driveResource->getSize(), $resourceMetadata['filesize'], "Expected file size to match, but they are different.");
+                $this->assertEquals($driveResource->getParent(), $resourceMetadata['file-parent'], "Expected file parent to match, but they are different.");
+                return;
+            }
+        }
+        $this->fail("Could not find file 'somefile.txt' personal the drive");
+    }
+
+    public function testEmptyFolderGetResourcesMetadata(): void
+    {
+        $resourceMetadata = $this->personalDrive->getResourceMetadata('/subfolder');
+        $driveResources = $this->personalDrive->getResources();
+        foreach ($driveResources as $driveResource) {
+            if ($driveResource->getName() === $resourceMetadata['name']) {
+                $this->assertEquals($driveResource->getId(), $resourceMetadata['id'], "Expected folder id to match, but they are different");
+                $this->assertEquals($driveResource->getSize(), $resourceMetadata['foldersize'], "Expected folder size to match, but they are different.");
+                $this->assertEquals($driveResource->getParent(), $resourceMetadata['file-parent'], "Expected folder parent to match, but they are different.");
+                return;
+            }
+        }
+        $this->fail("Could not find folder 'subfolder' inside personal drive");
+    }
+
+    /**
+     * @dataProvider invalidResources
+     */
+    public function testNonExistingResourceGetResourcesMetadata(string $invalidResources): void
+    {
+        $this->expectException(NotFoundException::class);
+        $this->personalDrive->getResourceMetadata($invalidResources);
+    }
+
+    public function testFolderGetResourcesMetadata(): void
```

Comment:
I think this test is :snake: :oil_drum: 
In a PROPFIND to a folder, the element 0 is the folder itself, that is thrown away in https://github.com/owncloud/ocis-php-sdk/pull/273/files#diff-326a1d6af0eef9fa2f105cb30f9476da1321c102f02aa3482b2d1983b8ff12e7R388
The new element 0 is the first child item. I believe `getResourceMetadata('/subfolder')` will get the properties of `uploaded.txt` or `innerfolder`
The test still passes because while cycling through the resources it comes to `uploaded.txt` and that is the same as `$resourceMetadata['name']`
I expect this test to fail when you do an explicit comparison as suggested by me above